src/app/core/profile/researcher-profile.service.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable max-classes-per-file */
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Operation, RemoveOperation, ReplaceOperation } from 'fast-json-patch';
+import { combineLatest, Observable, of as observableOf } from 'rxjs';
+import { catchError, find, map, switchMap, tap } from 'rxjs/operators';
+import { environment } from '../../../environments/environment';
+import { NotificationsService } from '../../shared/notifications/notifications.service';
+import { dataService } from '../cache/builders/build-decorators';
+import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { ConfigurationDataService } from '../data/configuration-data.service';
+import { DataService } from '../data/data.service';
+import { DefaultChangeAnalyzer } from '../data/default-change-analyzer.service';
+import { ItemDataService } from '../data/item-data.service';
+import { RemoteData } from '../data/remote-data';
+import { RequestService } from '../data/request.service';
+import { ConfigurationProperty } from '../shared/configuration-property.model';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { Item } from '../shared/item.model';
+import { NoContent } from '../shared/NoContent.model';
+import {
+  getFinishedRemoteData,
+  getFirstCompletedRemoteData,
+  getFirstSucceededRemoteDataPayload
+} from '../shared/operators';
+import { ResearcherProfile } from './model/researcher-profile.model';
+import { RESEARCHER_PROFILE } from './model/researcher-profile.resource-type';
+import { HttpOptions } from '../dspace-rest/dspace-rest.service';
+import { PostRequest } from '../data/request.models';
+import { hasValue } from '../../shared/empty.util';
+import {CoreState} from '../core-state.model';
+
+/**
+ * A private DataService implementation to delegate specific methods to.
+ */
+class ResearcherProfileServiceImpl extends DataService<ResearcherProfile> {
+    protected linkPath = 'profiles';
+
+    constructor(
+      protected requestService: RequestService,
+      protected rdbService: RemoteDataBuildService,
+      protected store: Store<CoreState>,
+      protected objectCache: ObjectCacheService,
+      protected halService: HALEndpointService,
+      protected notificationsService: NotificationsService,
+      protected http: HttpClient,
+      protected comparator: DefaultChangeAnalyzer<ResearcherProfile>) {
+      super();
+    }
+
+}
+
+/**
+ * A service that provides methods to make REST requests with researcher profile endpoint.
+ */
+@Injectable()
+@dataService(RESEARCHER_PROFILE)
+export class ResearcherProfileService {
```

Comment:
I might be missing it, but I think this entire file is missing specs.  I don't see a corresponding `researcher-profile.service.spec.ts

Reply:
done